src/main/java/it/pagopa/pm/gateway/service/VposService.java

```diff
@@ -198,7 +198,8 @@ private StepZeroResponse createStepZeroResponse(String errorMessage, String requ
     }
 
     private PaymentRequestEntity createEntity(String clientId, String mdcFields, String idTransaction, StepZeroRequest request) throws JsonProcessingException {
-        String requestJson = objectMapper.writeValueAsString(request);
+        VposDatabaseRequest vposDatabaseRequest = generateVposDatabaseRequest(request);
+        String requestJson = objectMapper.writeValueAsString(vposDatabaseRequest);
```

Comment:
Another refactor that would be nice to have would be to have `PaymentRequest` have overloads for `setJsonRequest` with a `setJsonRequest(VposPersistableRequest req)` signature so that manual marshalling is not necessary and the API is more ergonomic (also, less margin of error if one doesn't pass a `VposDatabaseRequest`)

Reply:
Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/211/commits/8274d3191fec5819b1123f93b1f62bb490dfd259